ebos/eclfluxmodule.hh

```diff
@@ -410,34 +410,61 @@ protected:
                                      unsigned timeIdx,
                                      const FluidState& exFluidState)
     {
+        const auto& scvf = elemCtx.stencil(timeIdx).boundaryFace(scvfIdx);
+        Scalar faceArea = scvf.area();
+        Scalar zEx = scvf.integrationPos()[dimWorld - 1];
         const auto& problem = elemCtx.problem();
+        calculateBoundaryGradients_(problem,
+                                    elemCtx.globalSpaceIndex(0, timeIdx),
+                                    elemCtx.intensiveQuantities(0, timeIdx),
+                                    scvfIdx,
+                                    timeIdx,
+                                    faceArea,
+                                    zEx,
+                                    exFluidState,
+                                    upIdx_,
+                                    dnIdx_,
+                                    volumeFlux_,
+                                    pressureDifference_);
+    }
+
+public:
+    /*!
+     * \brief Update the required gradients for boundary faces
+     */
+    template <class Problem, class FluidState, class EvaluationContainer>
+    static void calculateBoundaryGradients_(const Problem& problem,
```

Comment:
Why exactly is this static?

Reply:
To be able to get these gradients without requiring to store ExtensiveQuantities objects.

Reply:
Well, not sure how much compilation time this takes (a static template function in a template class ). If `TypeTag` is not needed here a free function might be better.

Reply:
I have now refactored to make it a free function. One minor issue is that we have to make some more types and enums from the Problem class public, actually I do not mind that, and just made them all public.

In the end I think it would be better as a static function though, primarily because of naming. If we are to make it a free function, we should give it a name specific to the flux module, i.e. `calculateBoundaryGradientsEclTrans` or similar. This is necessary to distinguish it from other flux modules that would otherwise also use the same function name, if refactored in a similar manner. This makes the calling code less modular, as instead of calling `ExtensiveQuantities::calculateBoundaryGradients()` it would call `calculateBoundaryGradientsEclTrans()` and lose the flexibility to use a different flux implementation. Therefore I think it is better to keep it inside the class that its behaviour is associated with.

For now, the last commit remains in the branch, for the purpose of discussion.

Reply:
I will kill the last commit for now. I'll stash it in a backup branch just in case.